common/model/transaction.h

```diff
@@ -94,6 +103,52 @@ struct _iota_transaction {
   bool solid;
 };
 
+struct _iota_transaction_meta_view_model {
```

Comment:
- Why not a `struct _iota_transaction` + metadata fields ?
- When do we use a `struct _iota_transaction_meta_view_model` instead of a `struct _iota_transaction` ?

Reply:
will remove `sapshot_index` and `solid` from `struct _iota_transaction`

Reply:
I really think that this `view model` naming from IRI is terrible. Let's call these models?
Also, the models don't have links to which tx they're actually referring to?

Reply:
you mean they should also contain the `hash`?

Reply:
at the cost of using more memory, why not have a parent struct that contains the views as materialised entries  + a bitflag that tells which of these were loaded?